### PR TITLE
Add the AWS guide

### PR DESCRIPTION
**Pull request by melvynator**
This is the pull request to add the AWS field guide. It contains instructions for the users that want to:

- Monitor their AWS infrastructure
- Monitor their AWS logs

https://observability-docs_610.docs-preview.app.elstc.co/guide/en/observability/master/monitor-aws.html

Closes https://github.com/elastic/observability-docs/issues/454

### CODE REVIEW

File: docs/en/observability/monitor-aws.asciidoc [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+Let's now configure another fileset and we are going to use the `s3access`
+one. The goal here is to be able to monitor how people access the bucket we
+created. This is what we achieved:
+
+image::one-bucket-archi.png[Current Architecture]
+
+Now what we need to do is to create another bucket and another queue. In order
+to create the following architecture:
+
+image::two-buckets-archi.png[Architecture with Access Logging Enabled]
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:303]: It seems like we're switching gears, but then we go back to what we've already achieved. What if we show the first screenshot before starting to switch context?

```suggestion
Here's what we've achieved so far:

image::one-bucket-archi.png[Current Architecture]

Now, let's configure the `s3access` fileset.
The goal here is to be able to monitor how people access the bucket we
created. To do this, we'll create another bucket and another queue.
The new architecture will look like this:

image::two-buckets-archi.png[Architecture with Access Logging Enabled]
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+
+Make sure the event notifications on the bucket you created is enabled and is
+sending notifications to the new queue you also just created. Now you need to go
+back to the first bucket you created, go to the properties, go to **Server access
+logging**, specify that you want to ship the access logs to the bucket you last
+created.
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:309]:
```suggestion
Create a new S3 bucket and SQS queue. Ensure that the event notifications on the new bucket are enabled,
and that it's sending notifications to the new queue.

Now go back to the first bucket, and go to *Properties* > *Server access
logging*. Specify that you want to ship the access logs to the bucket you most recently
created.
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+  elb:
+    enabled: false <1>
+  s3access:
+    enabled: true <5>
+    var.credential_profile_name: fb-aws <3>
+    var.queue_url: https://sqs.eu-central-1.amazonaws.com/836370109380/access-log <5>
+  vpcflow:
+    enabled: false <1>
+----
+<1> Disables the filesets.
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:336]: Similar thought here in regards to removing the <1>s

File: docs/en/observability/monitor-aws.asciidoc [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+[discrete]
+==== Step 6: Visualizing Logs
+
+Now that the logs are being shipped to Elasticsearch we can visualize them in
+Kibana. Go to Kibana, in the interface click on the left panel, click on
+**Logs** to see how your raw logs look like:
+
+image::EC2-logs.png[EC2 logs in the Logs UI]
+
+The filesets we used in the previous steps also comes with prebuilt dashboard
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:360]:
```suggestion
The filesets we used in the previous steps also come with prebuilt dashboards
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+----
+./filebeat -e
+----
+
+[discrete]
+==== Step 6: Visualizing Logs
+
+Now that the logs are being shipped to Elasticsearch we can visualize them in
+Kibana. Go to Kibana, in the interface click on the left panel, click on
+**Logs** to see how your raw logs look like:
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:356]: Changing this to be consistent with other Kibana terminology

```suggestion
Kibana. To see the raw logs, open the main menu in Kibana, then click
**Logs**:
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 6]
```diff
@@ -0,0 +1,596 @@
...
+
+image::EC2-logs.png[EC2 logs in the Logs UI]
+
+The filesets we used in the previous steps also comes with prebuilt dashboard
+that you can use to visualize the data. In Kibana, on the left panel, click on
+**dashboard**, search for S3 and select the dashboard called:
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:362]:
```suggestion
that you can use to visualize the data. In Kibana, open the main menu and click
**Dashboard**. Search for S3 and select the dashboard called:
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 7]
```diff
@@ -0,0 +1,596 @@
...
+**[Filebeat AWS] S3 Server Access Log Overview**:
+
+image::S3-Server-Access-Logs.png[S3 Server Access Log Overview]
+
+This give you an overview of how your S3 buckets are being accessed.
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:367]:
```suggestion
This gives you an overview of how your S3 buckets are being accessed.
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 8]
```diff
@@ -0,0 +1,596 @@
...
+==== Step 8: Install {metricbeat}
+
+On a new terminal window, download and install {metricbeat}.
+
+include::{beats-repo-dir}/tab-widgets/install-widget-metricbeat.asciidoc[]
+
+[discrete]
+==== Set up the assets
+
+Similarly to {filebeat}, {metricbeat} comes with predefined assets for parsing,
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:394]:
```suggestion
Similar to {filebeat}, {metricbeat} comes with predefined assets for parsing,
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 9]
```diff
@@ -0,0 +1,596 @@
...
+
+[discrete]
+==== Step 9: Configuring the {aws} module
+
+To collect the metrics from your {aws} infrastructure we will use the
+{metricbeat-ref}/metricbeat-module-aws.html[{aws}] module. The {aws} module is
+containing a lot of metricsets: `billing`, `cloudwatch`, `dynamodb`, `ebs`,
+`ec2`, `elb`, `lambda` and many more. Each metricsets is created to help you
+stream and process your data. In this tutorial we are going to show you a
+few examples using the `ec2` and the `billing` metricsets.
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:503]:
```suggestion
To collect metrics from your {aws} infrastructure, we'll use the
{metricbeat-ref}/metricbeat-module-aws.html[{metricbeat} {aws}] module. This module
contains many metricsets: `billing`, `cloudwatch`, `dynamodb`, `ebs`,
`ec2`, `elb`, `lambda`, and many more. Each metricset is created to help you
stream and process your data. In this tutorial, we're going to show you a
few examples using the `ec2` and the `billing` metricsets.
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 10]
```diff
@@ -0,0 +1,596 @@
...
+                "iam:ListAccountAliases",
+                "tag:getResources",
+                "ce:GetCostAndUsage"
+            ],
+            "Effect": "Allow",
+            "Resource": "*"
+        }
+    ]
+}
+----
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:564]: Oh snap! Where is this for Filebeat? This is exactly what I needed above

File: docs/en/observability/monitor-aws.asciidoc [hunk 11]
```diff
@@ -0,0 +1,596 @@
...
+./metricbeat -e
+----
+
+[discrete]
+==== Step 10: Visualizing Metrics
+
+Now that the metrics are being streamed to Elasticsearch we can visualize them in
+Kibana. Go to Kibana, in the interface click on the left panel, click on
+**Metrics** to see how your infrastructure look like. Once in the **Metrics UI**
+make sure to show the **{aws}** source and the **EC2 Instances**:
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:579]:
```suggestion
Kibana. In Kibana, open the main menu and click
**Metrics**. Make sure to show the **{aws}** source and the **EC2 Instances**:
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 12]
```diff
@@ -0,0 +1,596 @@
...
+
+image::EC2-instances.png[Your EC2 Infrastructure]
+
+The metricsets we used in the previous steps also comes with prebuilt dashboard
+that you can use to visualize the data. In Kibana, on the left panel, click on
+**dashboard**, search for EC2 and select the dashboard called: **[Metricbeat
```
Comment (bmorelli25) [inline docs/en/observability/monitor-aws.asciidoc:585]:
```suggestion
that you can use to visualize the data. In Kibana, open the main menu and click
**Dashboard**. Search for EC2 and select the dashboard called: **[Metricbeat
```

**Review by bmorelli25 — CHANGES_REQUESTED** [review]
This is looking great! I was able to create a new AWS account and follow along with all of the steps. I have a lot of small suggestions for improvement, mostly related to grammar and small order changes.

I also have these two questions. Take these with a grain of salt as this is my first time trying out SQS and the AWS module.

1. In step two I ran into trouble because my S3 bucket and my SQS queue were deployed in different regions. I couldn't figure out how to create an SQS queue in a region other than EU-CENTRAL, so I finally just had to delete my Oregon S3 and create a new S3 in EU-CENTRAL. Should we mention that they have to be in the same region?

2. In step 5, I get the following error: `awss3/collector.go:101	SQS ReceiveMessageRequest failed: EC2RoleRequestError: no EC2 instance role found`. It wasn't clear to me that the `fb-aws` role needs to be defined. But upon rereading, you do link to documentation about it. I cannot for the life of me understand what that AWS documentation was trying to tell me to do... CLI commands? But where do I use them? I finally found this link: `https://console.aws.amazon.com/iam/home?#/users` which I think would be useful to document. From there I just started guessing at which permission policies were needed. I was finally able to connect after adding the following permissions:
Screen Shot 2021-05-04 at 11 54 29 AM

I also had to add additional fields to the aws.yml config:

```
  ec2:
    enabled: true
    var.credential_profile_name: test-user
    var.access_key_id: key_id
    var.secret_access_key: my_access_key
    var.queue_url: my_url
```

EDIT: It looks like you included permissions for metricbeat. See https://github.com/elastic/observability-docs/pull/610#discussion_r626040359.

**Comment by melvynator** [comment]
Thanks for the review. 

> In step two I ran into trouble because my S3 bucket and my SQS queue were deployed in different regions. I couldn't figure out how to create an SQS queue in a region other than EU-CENTRAL, so I finally just had to delete my Oregon S3 and create a new S3 in EU-CENTRAL. Should we mention that they have to be in the same region?

Yeah, you are right I will add it to the guide. 

> It wasn't clear to me that the fb-aws role needs to be defined. But upon rereading, you do link to documentation about it.

I will try to come up with the same set of minimal permission that I defined for Metricbeat. Will that solve your issue?

**Comment by bmorelli25** [comment]
> I will try to come up with the same set of minimal permission that I defined for Metricbeat. Will that solve your issue?

Yes, it would. But only if you think it's useful to add. I'm new to the world of AWS, so I'm not sure what users would expect here.

File: docs/en/observability/monitor-aws.asciidoc (hunk 10 above)
Comment (melvynator) [reply docs/en/observability/monitor-aws.asciidoc:575]: I found the following minimal permission:

```
s3:GetObject
sqs:ReceiveMessage
sqs:ChangeMessageVisibility
sqs:DeleteMessage
```

File: docs/en/observability/monitor-aws.asciidoc [hunk 13]
```diff
@@ -0,0 +1,596 @@
...
+        }
+      ]
+    }
+  }
+}
+----
+
+<1> {filebeat} needs extra cluster permissions to publish logs, which differs
+from the {metricbeat} configuration used previously. You can find more
+details {filebeat-ref}/feature-roles.html[here].
```
[the remark this answers is not in the document]
Comment (melvynator) [reply docs/en/observability/monitor-aws.asciidoc:200]: I removed it.

File: docs/en/observability/monitor-aws.asciidoc [hunk 14]
```diff
@@ -0,0 +1,596 @@
...
+    var.credential_profile_name: fb-aws <3>
+    var.queue_url: https://sqs.eu-central-1.amazonaws.com/836370109380/howtoguide-tutorial <4>
+  elb:
+    enabled: false <1>
+  s3access:
+    enabled: false <1>
+  vpcflow:
+    enabled: false <1>
+----
+<1> Disables the filesets.
```
[the remark this answers is not in the document]
Comment (melvynator) [reply docs/en/observability/monitor-aws.asciidoc:291]: I removed the ones

File: docs/en/observability/monitor-aws.asciidoc (hunk 3 above)
Comment (melvynator) [reply docs/en/observability/monitor-aws.asciidoc:350]: I removed the ones

**Comment by melvynator** [comment]
@bmorelli25 I finished reviewing all the suggested edits. One small note with regards to the security settings:

```
 ec2:
    enabled: true
    var.credential_profile_name: test-user
    var.access_key_id: key_id
    var.secret_access_key: my_access_key
    var.queue_url: my_url
```

The setting `credential_profile_name` will not be used if you populate the `access_key_id` and `secret_access_key`. The `credential_profile_name` is the recommended way to use AWS credentials. This setting is actually using the AWS credentials stored on your computer in `~/.aws/credentials` this file can be formatted in a way were you can define different profiles with their different credential. A little bit like this:

```
[default]
aws_access_key_id=AKIAIOSFODNN7EXAMPLE
aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY

[user1]
aws_access_key_id=AKIAI44QH8DHBEXAMPLE
aws_secret_access_key=je7MtGbClwBF/2Zp9Utk/h3yCo8nvbEXAMPLEKEY
```

**Comment by bmorelli25** [comment]
> This setting is actually using the AWS credentials stored on your computer in `~/.aws/credentials` this file can be formatted in a way were you can define different profiles with their different credential.

Ahh! That makes so much more sense now. Let me give that method a try and take one more quick look at everything.

**Review by bmorelli25 — APPROVED** [review]
LGTM 🎉 

Let me know if you'd like me to take care of the backports for you.

**Pull request merged**